framework/wazuh/core/tests/test_utils.py

```diff
@@ -1998,3 +2000,68 @@ def test_agents_allow_higher_versions(new_conf, agents_conf):
+
+    mko = mock_open()
+    filename = "file.xml"
+    with patch('wazuh.core.utils.open', mko):
+        # Test upload simple content.
+        content = "Test"
+        with patch('wazuh.core.utils.safe_move') as mock_safe_move:
+            result = utils.upload_file(content, file_path=filename, check_xml_formula_values=False)
+            handle = mko()
+            assert isinstance(result, WazuhResult)
+            handle.write.assert_called_once_with(content)
+
+            tmp_path = os.path.join(OSSEC_TMP_PATH, 'file.tmp')
+            file_path = os.path.join(WAZUH_PATH, filename)
+            mock_safe_move.assert_called_once_with(tmp_path,
+                                                          file_path,
+                                                          ownership=(mock_uid(), mock_gid()),
+                                                          permissions=0o660)
+
+    mko = mock_open()
+    with patch('wazuh.core.utils.open', mko):
+        # Test upload formula content check_xml=True.
+        content = "<Test>+Tes't</Test>"
+        with patch('wazuh.core.utils.safe_move') as mock_safe_move:
+            result = utils.upload_file(content, file_path=filename, check_xml_formula_values=True)
+            handle = mko()
+            assert isinstance(result, WazuhResult)
+            handle.write.assert_called_once_with("<Test>'+Tes't</Test>")
+
+            tmp_path = os.path.join(OSSEC_TMP_PATH, 'file.tmp')
+            file_path = os.path.join(WAZUH_PATH, filename)
+            mock_safe_move.assert_called_once_with(tmp_path,
+                                                          file_path,
+                                                          ownership=(mock_uid(), mock_gid()),
+                                                          permissions=0o660)
+
+    # Test error 1005.
+    mko = mock_open()
+    with patch('wazuh.core.utils.open', mko):
+        with patch('wazuh.core.utils.safe_move') as mock_safe_move:
+            with pytest.raises(utils.WazuhInternalError, match=r'\b1005\b'):
+                mko().write.side_effect = IOError()
+                result = utils.upload_file(content, file_path=filename, check_xml_formula_values=False)
+
+    # Test error 1016.
+    mko = mock_open()
+    with patch('wazuh.core.utils.open', mko):
+        with patch('wazuh.core.utils.safe_move', side_effect=Error()) as mock_safe_move:
+            with pytest.raises(utils.WazuhInternalError, match=r'\b1016\b'):
+                result = utils.upload_file(content, file_path=filename, check_xml_formula_values=False)
+
+    # Test error 1006.
+    mko = mock_open()
+    with patch('wazuh.core.utils.open', mko):
+        with patch('wazuh.core.utils.safe_move', side_effect=PermissionError()) as mock_safe_move:
+            with pytest.raises(utils.WazuhError, match=r'\b1006\b'):
+                result = utils.upload_file(content, file_path=filename, check_xml_formula_values=False)
```

Comment:
We usually use a different function with the name  `<function_name>_ko` to test expected errors. 

Also, I think it would be appropiate to use parameters to force the different cases instead of repeating code and reassigning variables, something like

```py
@pytest.mark.parametrize('error_number, internal_error, mock_open_side_effect, safe_move_side_effect', [
    ('1015', True, IOError(), None),
    ('1016', True, None, Error()),
    ('1006', False, None, PermissionError())
])
```

Reply:
The new test_upload_file_ko function was created. All error test cases were moved to this function, which was also parameterized to avoid duplicated code.